src/basic/missing.h

```diff
@@ -537,12 +537,20 @@ struct btrfs_ioctl_quota_ctl_args {
 #  define DRM_IOCTL_DROP_MASTER _IO('d', 0x1f)
 #endif
 
-#if defined(__i386__) || defined(__x86_64__)
-
-/* The precise definition of __O_TMPFILE is arch specific, so let's
- * just define this on x86 where we know the value. */
+/* The precise definition of __O_TMPFILE is arch specific; use the
+ * values defined by the kernel:
+ * - alpha, parisc, sparc: all have a specific value;
+ * - others: they use the "generic" value.
+ */
 
 #ifndef __O_TMPFILE
+#if defined(__alpha__)
+#define __O_TMPFILE     0100000000
+#elif defined(__parisc__) || defined(__hppa__)
+#define __O_TMPFILE     0400000000
+#elif defined(__sparc__) || defined(__sparc64__)
+#define __O_TMPFILE     0x2000000
+#else
 #define __O_TMPFILE     020000000
 #endif
```

Comment:
Lennart,

As far as I can see, the generic one is octal `020000000` (4194304) and
the sparc one is hexa `0x2000000` (33554432). So they are different, no?

I also checked: alpha is octal `0100000000` (16777216) and hppa is octal
`0400000000` (67108864). So they are all different.

Unless I missed something...

(Note: I did duplicate the exact values as defined in the kernel; any
discrepancy is because of that.)

Yet, even if any two were identical, the kernel defines them specifically
for those three archs, so I believe it would still be better to mirror
the definitons here too. (But that's moot as they are different, and
unlikelly to change any time soon.)

Regards,
Yann E. MORIN.